game-app/game-core/src/main/java/games/strategy/triplea/TripleA.java

```diff
@@ -59,7 +59,9 @@ public void startGame(
       }
     }
     final LocalPlayers localPlayers = new LocalPlayers(players);
-    game.setDisplay(launchAction.startGame(localPlayers, game, players, chat));
-    game.setSoundChannel(launchAction.getSoundChannel(localPlayers));
+    LaunchAction.UserInterface userInterface =
+        launchAction.startGame(localPlayers, game, players, chat);
+    game.setDisplay(userInterface.getDisplay());
+    game.setSoundChannel(userInterface.getSound());
```

Comment:
`UserInterface` strikes me as a bit of an odd interface name. It is not bad, so I am a bit torn, but it seems somewhat overly generic for what it is - a tuple object.

I wonder if this block would get simpler if `game` were mutated and the end of the `startGame` method simply sets the game sound and display values. Side-effects are not good, but there are a lot of other side effects happening here that I think we could have a net-gain.

WDYT to move the `game.set` calls into the `startGame` method?

Reply:
I haven't thought about that. Mixed feelings on this. As you said side-effects are rarely intuitive, but they happen here anyways and I agree it would simplify the code. I think I'll go with that.